### PR TITLE
webgui: Added ability to set warning percentage for CPU, mempools fro…

### PR DESCRIPTION
**Pull request by rpenziol**
…m device edit page (#5895)

	*Set health progress bars for store, CPU, mempools to reach red threshold based on warning percentage

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

File: build.sql [hunk 1]
```diff
@@ -43,7 +43,7 @@ CREATE TABLE IF NOT EXISTS `processors` ( `processor_id` int(11) NOT NULL AUTO_I
...
-CREATE TABLE IF NOT EXISTS `storage` ( `storage_id` int(11) NOT NULL AUTO_INCREMENT, `device_id` int(11) NOT NULL, `storage_mib` varchar(16) NOT NULL, `storage_index` int(11) NOT NULL, `storage_type` varchar(32) DEFAULT NULL, `storage_descr` text NOT NULL, `storage_size` bigint(20) NOT NULL, `storage_units` int(11) NOT NULL, `storage_used` bigint(20) NOT NULL, `storage_free` bigint(20) NOT NULL, `storage_perc` text NOT NULL, `storage_perc_warn` int(11) DEFAULT '60', PRIMARY KEY (`storage_id`), KEY `device_id` (`device_id`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8;
+CREATE TABLE IF NOT EXISTS `storage` ( `storage_id` int(11) NOT NULL AUTO_INCREMENT, `device_id` int(11) NOT NULL, `storage_mib` varchar(16) NOT NULL, `storage_index` int(11) NOT NULL, `storage_type` varchar(32) DEFAULT NULL, `storage_descr` text NOT NULL, `storage_size` bigint(20) NOT NULL, `storage_units` int(11) NOT NULL, `storage_used` bigint(20) NOT NULL, `storage_free` bigint(20) NOT NULL, `storage_perc` text NOT NULL, `storage_perc_warn` int(11) DEFAULT '90', PRIMARY KEY (`storage_id`), KEY `device_id` (`device_id`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8;
```
Comment (laf) [inline build.sql]: I'm not personally a fan of 90 as the default, think it should be more 75 or so.

**Comment by laf** [comment]
Absolutely brilliant, thanks for contributing this :)

I'm not sure I'd agree the defaults for all of this should be 90%, it's too close to the max to be potentially useful. 60% or upto 75% seems more appropriate imho ( @librenms/reviewers ?)

The other alternative is we have a $config array that we use to set this and people can override:

```
$config['health']['limits']['storage'] = array(
    'high' => 90,
); 
$config['health']['limits']['memory'] = array(
    'high' => 90,
);
```

**Comment by rpenziol** [comment]
Thanks @laf ! The CPU warning level defaulted to 60% before. I changed it to 90% to match the normal 'red' colored progress bar threshold. In hindsight I probably shouldn't have touched that value. Keeping it at 60% to not mess up anyone's existing settings sounds like the best course of action.

File: build.sql (hunk 1 above)
Comment (murrant) [inline build.sql]: 80 sounds good to me.
I'm ok with making this change only affect new installs.

**Comment by murrant** [comment]
trigger ci

**Comment by laf** [comment]
@rpenziol are you updating this to revert the default back to 60%?

**Comment by rpenziol** [comment]
@laf Yes, I will revert it and push my changes. I'll try to push my changes either tomorrow or Monday

**Comment by laf** [comment]
Ace, thanks :)

**Comment by laf** [comment]
You've still got some defaults of 90 in there and imho a default of 90 is too high.

@librenms/reviewers anyone else?

**Comment by crcro** [comment]
it think a value around 70-75% should be more eloquent then 60 or 90 ...

**Comment by rpenziol** [comment]
@laf @crcro My motivation to setting it to 90% is that this value will determine the threshold on progress bars for CPU/mempool usage to turn bright red. Currently that threshold is 90% for all progress bars. The thought being that if this is lowered it may alarm users - as many more progress bars will be bright red than before. I also realize people will be able to set alerts based off of these new values, and 90% could be considered too high for many. I have no problem changing it per your recommendations.

**Comment by laf** [comment]
It's my opinion (and crcro as well) so I'd be interested in other opinions ( @librenms/reviewers ) but yes, 75% would be an acceptable value.

I realise this may show more red visually but the alerts are what are more important as those are automated, the former relies on people actually browsing to those specific pages.

**Comment by rpenziol** [comment]
@laf I don't have an objection with 75%. I'll wait for more opinions or until you give the go-ahead to update my pull request with an agreed-upon percentage.

**Comment by geordish** [comment]
75% sounds good to me.

File: html/includes/functions.inc.php [hunk 2]
```diff
@@ -190,10 +190,16 @@ function generate_overlib_content($graph_array, $text)
 }//end generate_overlib_content()
 
 
-function get_percentage_colours($percentage)
+function get_percentage_colours($percentage, $component_perc_warn = null)
 {
+    $perc_warn = '90';
```
Comment (laf) [inline html/includes/functions.inc.php]: I think just this one to adjust

**Pull request merged**